Doc/library/typing.rst

```diff
@@ -3109,16 +3110,21 @@ This module defines several deprecated aliases to pre-existing
 standard library classes. These were originally included in the typing
 module in order to support parameterizing these generic classes using ``[]``.
 However, the aliases became redundant in Python 3.9 when the
-corresponding pre-existing classes were enhanced to support ``[]``.
+corresponding pre-existing classes were enhanced to support ``[]`` (see
+:pep:`585`).
 
-The redundant types are deprecated as of Python 3.9 but no
-deprecation warnings are issued by the interpreter.
-It is expected that type checkers will flag the deprecated types
-when the checked program targets Python 3.9 or newer.
+The redundant types are deprecated as of Python 3.9. However, while the aliases
+may be removed at some point, removal of these aliases is not currently
+planned. As such, no deprecation warnings are currently issued by the
+interpreter for these aliases.
 
-The deprecated types will be removed from the :mod:`typing` module
-no sooner than the first Python version released 5 years after the release of Python 3.9.0.
-See details in :pep:`585`—*Type Hinting Generics In Standard Collections*.
+If at some point it is decided to remove these deprecated aliases, a
+deprecation warning will be issued by the interpreter for at least two releases
+prior to removal. The aliases are guaranteed to remain in the typing module
+without deprecation warnings until at least Python 3.14.
```

Comment:
Aha, so these are "soft deprecated".

https://peps.python.org/pep-0387/#soft-deprecation

It's a new term added in https://github.com/python/peps/pull/3182.

Would referencing that help add clarity?

Reply:
I'm honestly not sure it would add clarity, to be honest :/ I feel like it might be just another cross-reference that people would have to look up if they're reading the docs.

I'm not aware of another module that has a similar situation where so much of the module is soft-deprecated. (In Python 3.13, 42 out of 101 items in `typing.__all__` are soft-deprecated!) So I think I'd rather keep a verbose description in the typing docs of how we'll be specifically handling the typing deprecations, and treat them as their own unique thing, without reference to the broader concept of "soft deprecation"

Reply:
Okay!

(`optparse` and `getopt` (and likely several other modules) are _100%_ soft deprecated :)

Reply:
> (`optparse` and `getopt` (and likely several other modules) are _100%_ soft deprecated :)

True!